src/main/java/com/beust/jcommander/JCommander.java

```diff
@@ -459,9 +460,30 @@ private static String pluralize(int quantity, String singular, String plural) {
             }
         }
 
+        //
+        // Expand combined options
+        //
+        if (options.allowCombinedShortOptions) {
+            vResult2 = expandCombinedShortOptions(vResult2);
+        }
+
         return vResult2.toArray(new String[vResult2.size()]);
     }
 
+    private List<String> expandCombinedShortOptions(List<String> args) {
+        List<String> vResult = Lists.newArrayList();
+        for (String arg : args) {
+            if (arg.length() > 1 && arg.charAt(0) == '-' && arg.charAt(1) != '-') {
```

Comment:
Before doing this unconditionally, shouldn't you first check a few things? Specifically:

- If there is an actual option called `-abc`, which means you probably don't want to append
- Before breaking `-abc` into `-a`, `-b` and `-c`, make sure that each of these options exists and is a boolean

Am I missing something?

Reply:
The split arguments still pass through the usual checks, so if one of the options doesn't exist or has the wrong number of arguments, an appropriate error message is displayed.

At the moment there's no check to make sure that there is no `-abc` option - I can add that if you think it is helpful, but this would inevitably lead to ambiguity (e.g. if there is an option `-abc` as well as options `-a`, `-b`, and `-c`). Instead I would suggest when combined arguments are enabled, only short arguments should be allowed to use the `-` prefix, and long arguments use the `--` prefix. This is in line with what Python's [argparse](https://docs.python.org/3/library/argparse.html) package does by default.